lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/CallDcmawAsyncCriHandlerTest.java

```diff
@@ -0,0 +1,167 @@
+    @Mock private ConfigService mockConfigService;
+    @Mock private IpvSessionService mockIpvSessionService;
+    @Mock private ClientOAuthSessionDetailsService mockClientOAuthSessionDetailsService;
+    @Mock private DcmawAsyncCriService mockDcmawAsyncCriService;
+    @Mock private CriStoringService mockCriStoringService;
+    @Mock private IpvSessionItem mockIpvSessionItem;
+    @Mock private VerifiableCredentialResponse mockVerifiableCredentialResponse;
+    @InjectMocks private CallDcmawAsyncCriHandler callDcmawAsyncCriHandler;
+
+    @BeforeEach
+    public void setUp() {
+        mockIpvSessionItem.setIpvSessionId("a-session-id");
+    }
+
+    @Test
+    void handleRequestShouldCallDcmawCriAndReturnJourneyNextIfNoBreachingCiReceived()
+            throws Exception {
+        when(mockIpvSessionService.getIpvSession("a-session-id")).thenReturn(mockIpvSessionItem);
+        when(mockClientOAuthSessionDetailsService.getClientOAuthSession(any()))
+                .thenReturn(clientOAuthSessionItem);
+        when(mockVerifiableCredentialResponse.getUserId()).thenReturn(TEST_USER_ID);
+        when(mockVerifiableCredentialResponse.getCredentialStatus())
+                .thenReturn(VerifiableCredentialStatus.PENDING);
+        when(mockDcmawAsyncCriService.startDcmawAsyncSession(
+                        any(String.class),
+                        any(ClientOAuthSessionItem.class),
+                        any(IpvSessionItem.class)))
+                .thenReturn(mockVerifiableCredentialResponse);
+
+        Map<String, Object> lambdaResult =
+                callDcmawAsyncCriHandler.handleRequest(input, mockContext);
+
+        verify(mockCriStoringService)
+                .storeCriResponse(
+                        eq(input),
+                        eq(DCMAW_ASYNC_CRI),
+                        any(String.class),
+                        eq(clientOAuthSessionItem));
+
+        InOrder inOrder = inOrder(mockIpvSessionService);
+        inOrder.verify(mockIpvSessionService).updateIpvSession(mockIpvSessionItem);
+        inOrder.verifyNoMoreInteractions();
+
+        assertEquals("/journey/next", lambdaResult.get("journey"));
+    }
+
+    @Test
+    void handleRequestShouldReturnJourneyErrorResponseIfMissingIpvSessionId() {
+        ProcessRequest inputWithoutSessionId = new ProcessRequest();
+
+        Map<String, Object> lambdaResult =
+                callDcmawAsyncCriHandler.handleRequest(inputWithoutSessionId, mockContext);
+
+        assertEquals("/journey/error", lambdaResult.get("journey"));
+        assertEquals(HttpStatus.SC_BAD_REQUEST, lambdaResult.get("statusCode"));
+        assertEquals(ErrorResponse.MISSING_IPV_SESSION_ID.getCode(), lambdaResult.get("code"));
+        assertEquals(
+                ErrorResponse.MISSING_IPV_SESSION_ID.getMessage(), lambdaResult.get("message"));
+    }
+
+    @Test
+    void handleRequestShouldReturnJourneyErrorResponseIfDcmawCriServiceThrows() throws Exception {
```

Comment:
This isn't really about what throws the exception, just that it's not a `HttpResponseExceptionWithErrorBody` exception.